lab23-kaylee/test/cowsay-controller-test.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+require('angular');
+require('angular-mocks');
+
+describe('testing CowsayController', function() {
+  beforeEach(() => {
+    angular.mock.module('cowsayApp');
+    angular.mock.inject(($rootScope, $controller) => {
+      this.$rootScope = $rootScope;
+      this.cowsayCtrl = new $controller('CowsayController');
+      this.cowsayCtrl.$onInit();
+    });
+  });
+
+  afterEach(() => this.$rootScope.$apply());
+
+  it('should have an initial state', () => {
+    expect(this.cowsayCtrl.title).toEqual('Welcome to Cowville');
+    expect(this.cowsayCtrl.history).toEqual([]);
+    expect(this.cowsayCtrl.history.length).toEqual(0);
+    expect(this.cowsayCtrl.current).toEqual('beavis.zen');
+  });
+
+  it('should have plenty of "cows" to choose from', () => {
+    expect(this.cowsayCtrl.cowfiles.length).toEqual(46);
+  });
+
+  it('speak function', () => {
+    this.cowsayCtrl.speak('I\'m not really a cow!');
+    expect(this.cowsayCtrl.spoken).toContain('I\'m not really a cow!');
+    expect(this.cowsayCtrl.history[0]).toContain('I\'m not really a cow!');
+    expect(this.cowsayCtrl.history.length).toEqual(1);
+  });
+
+  it('update function', () => {
+    let updated = this.cowsayCtrl.update('Just kidding, I\'m actually a cow.');
+    expect(updated).toContain('Just kidding, I\'m actually a cow.');
+    // expect(this.cowsayCtrl.spoken).toEqual(updated);
+  });
+
+  it('undo function', () => {
+    this.cowsayCtrl.speak('hello');
+    expect(this.cowsayCtrl.history[0]).toContain('hello');
+    expect(this.cowsayCtrl.history.length).toEqual(1);
+    expect(this.cowsayCtrl.spoken).toContain('hello');
+
+    this.cowsayCtrl.speak('goodbye');
+    expect(this.cowsayCtrl.history[1]).toContain('goodbye');
+    expect(this.cowsayCtrl.history.length).toEqual(2);
+    expect(this.cowsayCtrl.spoken).toContain('goodbye');
+
+    this.cowsayCtrl.undo();
+    expect(this.cowsayCtrl.history[1]).toBeUndefined();
+    expect(this.cowsayCtrl.history.length).toEqual(1);
+    expect(this.cowsayCtrl.spoken).toContain('hello');
+  });
+});
+
+//Why does this.current get set to cows[0] ('beavis.zen') when cowsayCtrl.$onInit is called, and not 'elephant' as specified in ng-init directive in index.html?
```

Comment:
1. Might need to check your syntax on that `ng-init` directive. I'm guessing there's something in there that's throwing it off and not validating the new value.